R/zchunk_LA132.industry.R

```diff
@@ -48,48 +48,160 @@ module_energy_LA132.industry_DISABLED <- function(command, ...) {
+      mutate(value = value.x - value.y) %>%
+      select(-value.x, -value.y) ->
+      L132.in_EJ_R_indgasproc_F_Yh
+
+    ## Refining: crude oil refining energy consumption is not derived based on the output and assumed IO coefs so it doesn't apply here
+    ## Refining: Electricity was taken off prior to scaling for end-use sectors, and in the IEA energy balances, biofuels are treated as primary energy
+    L122.in_EJ_R_refining_F_Yh %>%
+      filter(!grepl("oil refining", sector)) %>%
+      filter(fuel %in% c("gas", "coal")) ->
+      L132.in_EJ_R_indrefining_F_Yh
+
+    # CHP: no adjustments necessary
+    L123.in_EJ_R_indchp_F_Yh -> L132.in_EJ_R_indchp_F_Yh
+
+    # Combine all of the deduction tables and multiply by -1 to indicate that these are deductions
+    L132.in_EJ_R_indunoil_F_Yh %>%
+      bind_rows(L132.in_EJ_R_indrefining_F_Yh) %>%
+      bind_rows(L132.in_EJ_R_indgasproc_F_Yh) %>%
+      bind_rows(L132.in_EJ_R_indchp_F_Yh) %>%
+      mutate(value = -1 * value) ->
+      L132.in_EJ_R_inddeductions_F_Yh
+
+    ## Heat: fuel inputs to heat need to be added to industrial energy use, in regions where heat is not modeled as a final fuel
+    # Calculate the share of heat consumed by the industrial sector, in regions where heat is not modeled as a separate fuel
+    L131.share_R_Senduse_heat_Yh %>%
+      filter(grepl("industry", sector)) %>%
+      select(-sector) %>%
+      group_by(GCAM_region_ID, fuel, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup ->
+      L132.share_R_indenergy_heat_Yh
+
+    # Multiply these shares by the energy inputs to heat
+    A_regions %>%
+      filter(has_district_heat == 0) %>%
+      select(GCAM_region_ID) %>%
+      unlist ->
+      has_district_heat_GCAM_region_ID
+
+    L124.in_EJ_R_heat_F_Yh %>%
+      filter(GCAM_region_ID %in% has_district_heat_GCAM_region_ID) %>%
+      left_join(L132.share_R_indenergy_heat_Yh, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(-fuel.y, -value.y, -value.x) %>%
+      rename(fuel = fuel.x) ->
+      L132.in_EJ_R_indheat_F_Yh
+
+    # Re-calculate industrial energy as original estimate minus fuel inputs to unconventional oil production, gas processing, and refining, and plus inputs to heat
+    L132.in_EJ_R_indenergy_F_Yh %>%
+      select(sector) %>%
+      unique %>%
+      unlist ->
+      temp_sector_value
+
+    L132.in_EJ_R_indenergy_F_Yh %>%
+      bind_rows(L132.in_EJ_R_inddeductions_F_Yh, L132.in_EJ_R_indheat_F_Yh) %>%
+      mutate(sector = temp_sector_value) ->
+      L132.in_EJ_R_Sindenergy_F_Yh
+
+    # Drop heat in regions where this fuel is backed out to its fuel inputs
+    A_regions %>%
+      filter(has_district_heat == 0) %>%
+      select(GCAM_region_ID) %>%
+      unique %>%
+      mutate(fuel = "heat") ->
+      region_heat
+
+    L132.in_EJ_R_Sindenergy_F_Yh %>%
+      anti_join(region_heat, by = c("GCAM_region_ID", "fuel")) %>%
```

Comment:
An explanatory comment would be good here: what's going on?

Reply:
Comments added.